File: domain/models/operation_log.go
package models

import (
	"time"

	"gorm.io/datatypes"
)

// OperationLog stores audit-style records for user/system actions.
type OperationLog struct {
	ID           uint           `gorm:"primaryKey;autoIncrement" json:"id"`
	UserID       *string        `gorm:"type:varchar(64);index" json:"user_id"`
	Module       string         `gorm:"type:varchar(100);index" json:"module"`
	Action       string         `gorm:"type:varchar(200)" json:"action"`
	API          string         `gorm:"type:varchar(255)" json:"api"`
	RequestData  datatypes.JSON `gorm:"type:json" json:"request_data"`
	Result       string         `gorm:"type:varchar(20);index" json:"result"`
	ErrorMessage *string        `gorm:"type:text" json:"error_message"`
	CreatedAt    time.Time      `gorm:"not null;autoCreateTime;index" json:"created_at"`
}

func (o *OperationLog) TableName() string {
	return "operation_logs"
}

File: web/src/types/operationLog.ts
export interface OperationLog {
  id: number
  user_id?: string
  module: string
  action: string
  api: string
  request_data?: Record<string, any>
  result: 'success' | 'failed'
  error_message?: string
  created_at: string
}

export interface OperationLogQuery {
  page: number
  page_size: number
  user_id?: string
  module?: string
  action?: string
  api?: string
  result?: string
  start_time?: string
  end_time?: string
}
